tiledb/ml/models/sklearn.py

```diff
@@ -39,30 +39,69 @@ def save(self, *, update: bool = False, meta: Optional[Meta] = None) -> None:
             target location.
         :param meta: Extra metadata to save in a TileDB array.
         """
+        if self.artifact is None:
+            raise RuntimeError("Model is not initialized")
+
         # Serialize model
         serialized_model = self._serialize_model()
 
         # Create TileDB model array
         if not update:
-            self.__create_array()
+            self._create_array(fields=["model"])
 
         self._write_array(serialized_model=serialized_model, meta=meta)
 
     def load(self, *, timestamp: Optional[Timestamp] = None) -> BaseEstimator:
         """
-        Load a Sklearn model from a TileDB array.
+        Load switch, i.e, decide between __load (TileDB-ML<=0.8.0) or __load_v2 (TileDB-ML>0.8.0).
 
+        Load a Sklearn model from a TileDB array.
         :param timestamp: Range of timestamps to load fragments of the array which live
             in the specified time range.
         :return: A Sklearn model object.
         """
         # TODO: Change timestamp when issue in core is resolved
 
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            # Check if we try to load models with the old 1-cell schema.
+            if model_array.schema.domain.size < np.iinfo(np.uint64).max - 1025:
```

Comment:
True story, and I don't like it either, but we don't keep tiledb-ml version somewhere at the moment. Any other suggestions on this?

Reply:
Start writing the ml version in this PR and check for its presence? 

If adding the version should be postponed to a separate PR, at least hide this condition in a (hopefully temporary) `is_v1` or `is_v2` function/method to avoid copy-pasting it 3 times.

Reply:
Ok, I will go with the second approach and add version in a following PR.